devito/mpi/halo_scheme.py

```diff
@@ -76,13 +76,16 @@ def __init__(self, exprs, ispace):
         # Track the IterationSpace offsets induced by SubDomains/SubDimensions.
         # These should be honored in the derivation of the `omapper`
         self._honored = {}
-        for i in ispace.intervals:
-            if not i.dim._defines & set(self.dimensions):
-                continue
-            elif i.dim.is_Sub and not i.dim.local:
-                ltk, _ = i.dim.thickness.left
-                rtk, _ = i.dim.thickness.right
-                self._honored[i.dim.root] = frozenset([(ltk, rtk)])
+        # SubDimensions are not necessarily included in directloy in
+        # ispace.dimensions and hence we need to first utilize the `_defines` method
+        dims = set().union(*[d._defines for d in ispace.dimensions])
+        dims.difference_update([d for d in ispace.dimensions if
```

Comment:
this line can probably be squashed with the previous one, by adding the negated if condition to the comprehension?

```
set().union(*[d._defines for d in ispace.dimensions if d._defines & self.dimensions])
```

also, cast of `self.dimensions` to `set` now unnecessary

Reply:
fixed.